domain/consensus/utils/subnetworks/from_string.go

```diff
@@ -8,13 +8,18 @@ import (
 
 // FromString creates a DomainSubnetworkID from the given byte slice
 func FromString(str string) (*externalapi.DomainSubnetworkID, error) {
-	runes := []rune(str)
-	for i := 0; i < externalapi.DomainSubnetworkIDSize*2; i++ {
-		runes[i], runes[externalapi.DomainSubnetworkIDSize-1-i] = runes[externalapi.DomainSubnetworkIDSize-1-i], runes[i]
+	subnetworkIDBytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
 	}
```

Comment:
I would first reverse the bytes, and only then call `FromBytes`

Reply:
But then I need to check the size and handle it if it's not even.